src/mips/psyqo/advancedpad.hh

```diff
@@ -143,7 +129,7 @@ class AdvancedPad {
      * @param pad The pad to query.
      * @return A boolean value indicating whether the pad is connected.
      */
-    bool isPadConnected(Pad pad) const { return (m_padData[pad][0] & 0xff) == 0; }
+    bool isPadConnected(Pad pad) const { return m_padData[static_cast<unsigned>(pad)].connected == 0; }
```

Comment:
It _might_ be prettier to use `std::to_underlying(pad)` instead of the static_cast throughout this file. It comes from `<utility>`, and it's been added in C++23.

Reply:
It's a bit shorter, but looks like we'd need to bump the whole SDK version for that change(currently on C++ 20). I'm ok with working around this with the casts for now if changing the minimum version is undesirable.

Reply:
_Ah_.

We could theoretically polyfill it somewhere in the codebase: https://godbolt.org/z/7r4T6nq3e

As you wish, really.

Reply:
I gave this a shot in the latest commit